tests/test_utils.py

```diff
@@ -413,3 +415,43 @@ def test_map_optional():
 
     assert map_optional(to_upper, None) is None
     assert map_optional(to_upper, "hello") == "HELLO"
+
+
+def test_get_file_reload_register_func_if_changed():
+    with tempfile.TemporaryDirectory(
```

Comment:
can't you use `tmp_path` fixture here too?

Reply:
yes, will change.